happ/zomes/projects_integrity/src/project/project_meta/entry.rs

```diff
@@ -2,6 +2,36 @@ use hdi::prelude::*;
 use holo_hash::{ActionHashB64, AgentPubKeyB64};
 use std::*;
 
+use crate::ui_enum::UIEnum;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, SerializedBytes)]
+#[serde(from = "UIEnum")]
+#[serde(into = "UIEnum")]
+pub enum LayeringAlgorithm {
+    LongestPath,
+    CoffmanGraham,
+}
+
+impl From<UIEnum> for LayeringAlgorithm {
+    fn from(ui_enum: UIEnum) -> Self {
+        match ui_enum.0.as_str() {
+            "LongestPath" => Self::LongestPath,
+            "CoffmanGraham" => Self::CoffmanGraham,
+            _ => Self::CoffmanGraham,
```

Comment:
this is probably what you meant by a default, but this would still only work if the property existed on the input object, and just was an empty string "" or some other string like "whatever". 
It's not ideal to have in a way, but Rust forces it.

Reply:
we could switch to using `if let ...` instead of a `match` statement, which would remove the `_ => Self::CoffmanGraham`, but i think its fine the way it is.